x-pack/filebeat/input/awss3/input.go

```diff
@@ -62,13 +94,55 @@ func (in *s3Input) Test(ctx v2.TestContext) error {
 }
 
 func (in *s3Input) Run(ctx v2.Context, pipeline beat.Pipeline) error {
-	collector, err := in.createCollector(ctx, pipeline)
+	var err error
+	var collector s3Collector
+
+	persistentStore, err := in.store.Access()
+	if err != nil {
+		return fmt.Errorf("Can not access persistent store: %w", err)
+	}
+
+	states := NewStates()
+	err = states.readStatesFrom(persistentStore)
+	if err != nil {
+		return fmt.Errorf("Can not start persistent store: %w", err)
+	}
+
+	metricRegistry := monitoring.GetNamespace("dataset").GetRegistry()
+	inputMetrics := newInputMetrics(metricRegistry, ctx.ID)
+
+	if in.config.QueueURL != "" {
+		collector, err = in.createSQSCollector(ctx, pipeline, inputMetrics)
+		if err != nil {
+			return fmt.Errorf("cannot create SQS collector: %w", err)
+		}
+	}
+
+	if in.config.S3Bucket != "" {
+		collector, err = in.createS3BucketCollector(ctx, pipeline, inputMetrics, states, persistentStore)
+		if err != nil {
+			return fmt.Errorf("cannot create S3 bucket collector: %w", err)
+		}
+	}
+
+	err = in.grp.Go(func(canceler unison.Canceler) error {
+		interval := in.store.CleanupInterval()
+		if interval <= 0 {
+			interval = 5 * time.Minute
+		}
+
+		interval = 1 * time.Minute
```

Comment:
this last line invalidates all the previous

Reply:
leftover from debugging